contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,398 @@
+
+```js
+// redirect-test/middleware.js
+//... rest of code
+
+	const requestOptions = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      api_key: ph_project_api_key,
+      distinct_id: distinct_id
+    })
+  };
+  
+  // Evaluate experiment flag
+  const ph_request = await fetch(
+    'https://app.posthog.com/decide?v=3', // or eu.posthog.com
+    requestOptions
+  );
+  const data = await ph_request.json();
+  const flagResponse = data.featureFlags['main-redirect']
+
+  // Redirect to correct page
+  if (request.nextUrl.pathname=== '/test' && flagResponse === 'control') {
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  if (request.nextUrl.pathname=== '/control' && flagResponse === 'test') {
+    return NextResponse.redirect(new URL('/test', request.url))
+  }
+  return NextResponse.next()
+}
+//... rest of code
+```
+
+### Capturing exposure
+
+To get accurate results for our experiment, we also need to capture a `$feature_flag_called` event after the feature flag has been evaluated, but before redirecting. This is known as an exposure event and shows a user is part of the experiment. It requires another POST request like this:
+
+```js
+//... rest of code, after flag evaluation
+
+// Capture exposure event
+const eventOptions = {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json'
+  },
+  body: JSON.stringify({
+    api_key: ph_project_api_key,
+    distinct_id: distinct_id,
+    properties: {
+      "$feature_flag": 'main-redirect',
+      "$feature_flag_response": flagResponse
+    },
+    event: "$feature_flag_called"
+  })
+};
+
+const eventRequest = await fetch(
+  'https://app.posthog.com/capture',
+  eventOptions
+);
+
+//... rest of code, redirect logic
+```
+
+## Bootstrapping the data
+
+The final piece to our redirect test is bootstrapping the user distinct ID (and flags while we are at it). Bootstrapping is adding the flag and user data from the server to the client library so it is available as soon as the client PostHog library loads.
+
+> **Why is bootstrapping necessary?** If we didn't bootstrap the distinct ID, PostHog would set a second distinct ID for the same user on the frontend. When calculating the results of the experiment, PostHog wouldn't know the two were connected, creating a broken test.
+
+We create a `bootstrapData` cookie with the flags and distinct ID data and then add it to the response. We also add a check for the `bootstrapData` cookie in the middleware when we are creating the distinct ID so we don’t get two different IDs whenever we redirect.
+
+When put together with everything else, our final `middleware.js` file looks like this:
```

Comment:
Just clarifying what bootstrapping means 
```suggestion
The final piece to our redirect test is [bootstrapping](/docs/feature-flags/bootstrapping) the user distinct ID and feature flags. Bootstrapping is when you initialize PostHog with precomputed user data so that it is available as soon as PostHog loads – without needing to make additional API calls 

> **Why is bootstrapping necessary?** If we didn't bootstrap the distinct ID, PostHog would set a second distinct ID for the same user on the frontend. When calculating the results of the experiment, PostHog wouldn't know the two were connected, creating a broken test.

We create a `bootstrapData` cookie with the flags and distinct ID data and then add it to the response. We also add a check for the `bootstrapData` cookie in the middleware when we are creating the distinct ID so we don’t get two different IDs whenever we redirect.

When put together with everything else, our final `middleware.js` file looks like this:
```